client/src/main/java/org/asynchttpclient/cookie/ThreadSafeCookieStore.java

```diff
@@ -15,18 +15,29 @@
 package org.asynchttpclient.cookie;
 
 import io.netty.handler.codec.http.cookie.Cookie;
+import io.netty.util.Timeout;
+import io.netty.util.TimerTask;
+
 import org.asynchttpclient.uri.Uri;
 import org.asynchttpclient.util.Assertions;
 import org.asynchttpclient.util.MiscUtils;
 
 import java.util.*;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
 public final class ThreadSafeCookieStore implements CookieStore {
+  private static final int COOKIE_REMOVAL_DELAY_IN_SECONDS = 30;
+
+  private final Map<String, Map<CookieKey, StoredCookie>> cookieJar = new ConcurrentHashMap<>();
+  private io.netty.util.Timer timer;
 
-  private Map<CookieKey, StoredCookie> cookieJar = new ConcurrentHashMap<>();
+  public ThreadSafeCookieStore(io.netty.util.Timer timer) {
```

Comment:
Again, why does the CookieStore need to be aware of the Timer that cleans it up? I think it should expose methods for an external timer to clean it up. As proof it would be a better design: it would make testing way easier.

Reply:
Sure. Let me fix this. Thanks for the feedback.